index.bs

```diff
@@ -2793,25 +2793,31 @@ Note: Invoking this method from a [=browsing context=] where the [=Web Authentic
 
 </div>
 
-### Availability of a [=passkey platform authenticator=] - PublicKeyCredential's `isPasskeyPlatformAuthenticatorAvailable()` Method ### {#sctn-isPasskeyPlatformAuthenticatorAvailable}
+### Availability of [=client=] capabilities - PublicKeyCredential's `getClientCapabilities()` Method ### {#sctn-getClientCapabilities}
 
-<div link-for-hint="WebAuthentication/isPasskeyPlatformAuthenticatorAvailable">
+<div link-for-hint="WebAuthentication/getClientCapabilities">
 
-[=[WRPS]=] use this method to determine whether they can create a new [=passkey=] using a [=user-verifying platform authenticator=] or a {{AuthenticatorTransport/hybrid}} authenticator.
-Upon invocation, the [=client=] employs a [=client platform=]-specific procedure to discover available [=user-verifying platform authenticators=] and the
-availability of {{AuthenticatorTransport/hybrid}} transport.
-If one or both are discovered, the promise is resolved with the value of [TRUE].
-If neither is discovered, the promise is resolved with the value of [FALSE].
-Based on the result, the [=[RP]=] can take further actions to guide the user to create a [=passkey=].
+[=[WRPS]=] use this method to determine the availability of a limited set of [=WebAuthn Client|client=] capabilities to offer certain workflows and experiences to users. For example, an RP may offer a sign in button on clients where only {{AuthenticatorTransport/hybrid}} transport is available or where {{CredentialMediationRequirement/conditional}} mediation is unavailable (instead of showing a username field).
 
-This method has no arguments and returns a Boolean value.
+Upon invocation, the [=client=] employs a [=client platform=]-specific procedure to discover availablity of these capabilities.
+
+This method has no arguments and returns a record of capability keys to Boolean values.
 
 <xmp class="idl">
     partial interface PublicKeyCredential {
-        static Promise<boolean> isPasskeyPlatformAuthenticatorAvailable();
+        static Promise<PublicKeyCredentialClientCapabilities> getClientCapabilities();
     };
+
+    typedef record<DOMString, boolean> PublicKeyCredentialClientCapabilities;
 </xmp>
 
+[=map/Keys=] in {{PublicKeyCredentialClientCapabilities}} MUST be sorted in ascending lexicographical order.
```

Comment:
I think this comment about sorting is moot with a `record`? We're now in the world of Javascript object order which is complex and not something that this should have an opinion on, I think.

Reply:
Makes sense to me, and I'd hazard to assume that no developer would go looking to make sure this is true. I've removed it.

Reply:
We added this [for the sake of minimizing fingerprinting opportunities](https://github.com/w3c/webauthn/pull/1923#discussion_r1341194415), not as something for developers to use - just like `getTransports()` is required to return a sorted list for the same reason. [Record types](https://webidl.spec.whatwg.org/#idl-record) are also defined to be [ordered maps](https://infra.spec.whatwg.org/#ordered-map) just like `maplike`s, so I don't think the requirement is moot. I think the order requirement should stay, but I don't feel very strongly about it.

Reply:
@emlun I'm even less opinionated here, I didn't consider order of keys being something you could use for fingerprinting. I'm happy to add it back in if it prevents us from having to revisit the user privacy discussion (which, to be clear, I thought was good and ended with us in a good spot.)

@agl when you mention that "JavaScript object order...is complex" was that from an implementation perspective? Does it put an undue burden on client developers if the sorting requirement is restored?

Reply:
In the name of maintaining the fingerprinting minimization aspects of this API, and because `record` and `maplike` are both ordered types I restored this line. I also alphabetized the `ClientCapability` enums to subconsciously encourage clients to lexicographically sort these values when they support this API 😏

Reply:
I thought that Javascript had defined the iteration order these days and that we shouldn't override that. Indeed [it has](https://tc39.es/ecma262/#sec-ordinaryownpropertykeys), but the defined order is partially based on the insertion order. So, yep, this should stay.